revangel/nostalgiabot2
# Add Precommit Hooks; Linting

# Overview
This PR adds [pre-commit](https://pre-commit.com/), a framework recommended by flake8 to manage git pre-commit hooks. With this, [flake-8](https://flake8.pycqa.org/en/latest/index.html) and [black](https://github.com/psf/black) are added as pre-commit hooks

The requirements files have also been refactored to implement a separate **dev requirements** file, which seemed appropriate for specifying the linting dependencies this PR uses.

# References
- [NB-41](https://github.com/revangel/nostalgiabot2/issues/41)

# Notes
- Configurations are kept minimal for the time being
- The **max line length** is 100, which was arbitrarily chosen because it seems like a nice number
- `W503` is ignored, because it's apparently [being revised to be the opposite of what it's warning against](https://www.flake8rules.com/rules/W503.html)
- Currently, if there are flake8 issues that black can't format automatically, the commits won't go through
  - Might want to address this in the future

# Testing Guidelines
- There is a **test linting file** that you can use to check the pre-commit hooks
- Make changes that will make the linter complain, commit, and see if those changes have been automatically cleaned up by black
